src/controllers/authController.js

```diff
@@ -65,8 +67,47 @@ const logout = async (req, res) => {
   res.status(StatusCodes.OK).json({ msg: 'user logged out!' });
 };
 
+//forgotPassword endpoint
+const forgotPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw new CustomError.BadRequestError('Please provide an email');
+  }
+
+  const user = await User.findOne({ email });
+
+  if (!user) {
+    throw new CustomError.NotFoundError('User not found');
+  }
+
+  // Generate a random token
+  const resetToken = crypto.randomBytes(20).toString('hex');
+  const resetExpires = Date.now() + 15 * 60 * 1000; // 15 minutes from now
+
+  // Update user's reset token and expiration date
+  user.passwordResetToken = resetToken;
+  user.passwordResetExpiresOn = new Date(resetExpires);
+  await user.save();
+
+  // Send reset password email
+  const emailTemplate = 'forgotPassword'; // to adjust this based on our email template name
+  const recipient = user.email;
+  const localVariables = {
+    resetToken,
+  };
+  await emailSender.sendForgotPasswordEmail(
+    emailTemplate,
```

Comment:
I wouldn't expect to need to pass in the template name to the `sendForgotPasswordEmail` function. Will different callers be passing different email templates to that function? I think it will always use the same template, so that should be just a static variable defined inside that function itself rather than passed in from the controller

Reply:
Could you check please ? commit  346fdf55e289cc9a37f919fedb7f46d0452b0ade

Reply:
Yes this line (155) that passes in `emailTemplate` can be removed. 

Your code in `sender.js` has `sendForgotPasswordEmail` only expecting two arguments: the recipient email address, and the reset token

Reply:
Done , commit 66413a776311332e85d5529378bcdeda744cd532

Reply:
I think you accidentally added it back in in the latest commit: b0d0cd7